test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -603,4 +605,140 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(flavorOnDemand).
+						Resource(corev1.ResourceCPU, "5", "5").Obj(),
+				).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
+		})
+
+		ginkgo.It("Should update of the pending workloads when a new workload is scheduled", func() {
+			const lowPrio, midPrio, highPrio = 0, 10, 100
+			workloadsFirstBatch := []*kueue.Workload{
+				testing.MakeWorkload("one", ns.Name).Queue(localQueue.Name).Priority(highPrio).
+					Request(corev1.ResourceCPU, "2").Request(resourceGPU, "2").Obj(),
+				testing.MakeWorkload("two", ns.Name).Queue(localQueue.Name).Priority(midPrio).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+
+			gomega.Eventually(func() *kueue.ClusterQueuePendingWorkloadsStatus {
+				var updatedCq kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCq)).To(gomega.Succeed())
+				return updatedCq.Status.PendingWorkloadsStatus
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&kueue.ClusterQueuePendingWorkloadsStatus{}, ignoreLastChangeTime))
+
+			ginkgo.By("Creating workloads")
+			for _, w := range workloadsFirstBatch {
+				gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+			}
+			gomega.Eventually(func() *kueue.ClusterQueuePendingWorkloadsStatus {
+				var updatedCq kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCq)).To(gomega.Succeed())
+				return updatedCq.Status.PendingWorkloadsStatus
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&kueue.ClusterQueuePendingWorkloadsStatus{
+				Head: []kueue.ClusterQueuePendingWorkload{
+					{
+						Name:      "one",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "two",
+						Namespace: ns.Name,
+					},
+				},
+			}, ignoreLastChangeTime))
+
+			ginkgo.By("Creating a new workload")
+			workloadsSecondBatch := []*kueue.Workload{
+				testing.MakeWorkload("three", ns.Name).Queue(localQueue.Name).Priority(midPrio).
+					Request(corev1.ResourceCPU, "2").Request(resourceGPU, "2").Obj(),
+				testing.MakeWorkload("four", ns.Name).Queue(localQueue.Name).Priority(lowPrio).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+			for _, w := range workloadsSecondBatch {
+				gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+			}
+
+			ginkgo.By("Cut off the head of pending workloads when the number of pending workloads exceeds MaxCount")
```

Comment:
```suggestion
			ginkgo.By("Verify the head of pending workloads when the number of pending workloads exceeds MaxCount")
```
Will sound better :)

Reply:
changed here https://github.com/kubernetes-sigs/kueue/pull/1069/commits/20448ee40ed7f91bb885cdca263f5e10fe1bbe5b